src/misc/cross.cpp

```diff
@@ -782,6 +783,91 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#ifndef WIN32
+uint16_t xattr_string_to_fattr(const char* xattr)
+{
+	uint16_t fattr = 0;
+
+	if (strchr(xattr, 'R')) {
+		fattr |= DOS_ATTR_READ_ONLY;
+	}
+	if (strchr(xattr, 'H')) {
+		fattr |= DOS_ATTR_HIDDEN;
+	}
+	if (strchr(xattr, 'S')) {
+		fattr |= DOS_ATTR_SYSTEM;
+	}
+	if (strchr(xattr, 'A')) {
+		fattr |= DOS_ATTR_ARCHIVE;
+	}
+
+	return fattr;
+}
+
+void fattr_to_xattr_string(uint16_t fattr, char* outbuf, size_t outsize)
+{
+	memset(outbuf, 0, outsize);
+
+	if (fattr & DOS_ATTR_READ_ONLY) {
+		strcat(outbuf, "R");
+	}
+	if (fattr & DOS_ATTR_HIDDEN) {
+		strcat(outbuf, "H");
+	}
+	if (fattr & DOS_ATTR_SYSTEM) {
+		strcat(outbuf, "S");
+	}
+	if (fattr & DOS_ATTR_ARCHIVE) {
+		strcat(outbuf, "A");
+	}
+}
+
+uint16_t get_fat_attribs_from_xattr(const char *path)
```

Comment:
Ideally, you'd use `std_fs::path` and only covert at the "edges", i.e. when you must convert it to a C string so you can pass it to an OS-level C function.